src/utils/static-server.ts

```diff
@@ -1,40 +1,36 @@
+import { createServer } from 'http'
+import { AddressInfo } from 'net'
 import path from 'path'
 
-import fastifyStatic from '@fastify/static'
-import Fastify from 'fastify'
+import { App, Request, Response } from '@tinyhttp/app'
+import sirv from 'sirv'
 
 import { log, NETLIFYDEVLOG } from './command-helpers.js'
 
-/**
- * @param {object} config
- * @param {import('./types.js').ServerSettings} config.settings
- */
-// @ts-expect-error TS(7031) FIXME: Binding element 'settings' implicitly has an 'any'... Remove this comment to see the full error message
-export const startStaticServer = async ({ settings }) => {
-  const server = Fastify()
-  const rootPath = path.resolve(settings.dist)
-  server.register(fastifyStatic, {
-    root: rootPath,
-    etag: false,
-    acceptRanges: false,
-    lastModified: false,
-  })
+export const startStaticServer = ({ settings }: { settings: import('./types.js').ServerSettings }) => {
+  const rootPath = path.resolve(settings.dist || '')
 
-  server.setNotFoundHandler((_req, res) => {
-    res.code(404).sendFile('404.html', rootPath)
+  const app = new App({
+    noMatchHandler: (req, res) => {
+      res.status(404).sendFile('400.html', { root: rootPath })
+    },
   })
+  app.use(sirv(rootPath, { etag: false }))
 
-  server.addHook('onRequest', (req, reply, done) => {
-    reply.header('age', '0')
-    reply.header('cache-control', 'public, max-age=0, must-revalidate')
+  app.use((req, res, next) => {
+    res.header('age', '0')
+    res.header('cache-control', 'public, max-age=0, must-revalidate')
     const validMethods = ['GET', 'HEAD']
-    if (!validMethods.includes(req.method)) {
-      reply.code(405).send('Method Not Allowed')
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    if (!validMethods.includes(req.method!)) {
+      res.status(405).send('Method Not Allowed')
     }
-    done()
+    return next?.()
   })
-  await server.listen({ port: settings.frameworkPort })
-  const [address] = server.addresses()
+
+  const server = createServer((req, res) => app.handler(req as Request, res as Response))
```

Comment:
Type assertions are unsafe. Why is there a mismatch here? How can we resolve it?

Reply:
tinyhttp's Request and Response extend `node:http`'s Request and Response. Probably could pass as a generic to `createServer` but if I recall correctly, it throws a type error

Reply:
this is a TypeScript issue. `Request` extends `IncomingMessage`, which itself extends `Stream`. TypeScript thinks that `Request` doesn't extends `Stream` because of that, even though it does